pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -3778,6 +3778,98 @@ var _ = Context("Scheduling", func() {
+						v1alpha1.LabelReservationID: fmt.Sprintf("r-%s", it.Name),
+					}),
+					Price: fake.PriceFromResources(it.Capacity) / 10000.0,
+				})
+			}
+		})
+		FIt("shouldn't fallback to on-demand or spot when compatible reserved offerings are available", func() {
+			// With the pessimistic nature of scheduling reservations, we'll only be able to provision one instance per loop if a
+			// nodeclaim is compatible with both instance types
+			cloudProvider.ReservationManagerProvider.SetCapacity("r-small-instance-type", 1)
+			cloudProvider.ReservationManagerProvider.SetCapacity("r-medium-instance-type", 1)
+			ExpectApplied(ctx, env.Client, nodePool)
+
+			pods := lo.Times(3, func(_ int) *corev1.Pod {
+				return test.UnschedulablePod(test.PodOptions{
+					ResourceRequirements: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							// Ensures that this can fit on both small and medium, but two can't fit on medium
+							corev1.ResourceCPU: resource.MustParse("1800m"),
+						},
+					},
+				})
+			})
+
+			// All pods won't be able to fit on a single small or medium instance, but we're not going to create a large instance
+			// since that would involve falling back to on-demand or spot. Instead, we'll schedule a single pod this loop. We
+			// can't schedule all three because we don't know what instance type will be selected in the launch flow, so the
+			// single nodeclaim reserves both the small and medium offerings.
+			bindings := ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pods...)
+			Expect(len(bindings)).To(Equal(1))
+			node := lo.Values(bindings)[0].Node
+			Expect(node.Labels).To(HaveKeyWithValue(v1.CapacityTypeLabelKey, v1.CapacityTypeReserved))
+			Expect(node.Labels).To(HaveKeyWithValue(corev1.LabelInstanceTypeStable, "small-instance-type"))
+
+			pods = lo.Filter(pods, func(p *corev1.Pod, _ int) bool {
+				return bindings.Get(p) == nil
+			})
+
+			// Again, we'll only be able to schedule a single pod
+			bindings = ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pods...)
+			Expect(len(bindings)).To(Equal(1))
+			node = lo.Values(bindings)[0].Node
+			Expect(node.Labels).To(HaveKeyWithValue(v1.CapacityTypeLabelKey, v1.CapacityTypeReserved))
+			Expect(node.Labels).To(HaveKeyWithValue(corev1.LabelInstanceTypeStable, "medium-instance-type"))
+
+			pods = lo.Filter(pods, func(p *corev1.Pod, _ int) bool {
+				return bindings.Get(p) == nil
+			})
+
+			bindings = ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pods...)
+			Expect(len(bindings)).To(Equal(1))
+			node = lo.Values(bindings)[0].Node
+			Expect(node.Labels).To(HaveKeyWithValue(v1.CapacityTypeLabelKey, Not(Equal(v1.CapacityTypeReserved))))
+			Expect(node.Labels).To(HaveKeyWithValue(corev1.LabelInstanceTypeStable, "small-instance-type"))
+		})
+		// It("should fallback to on-demand or spot when no compatible reserved offerings are available", func() {
```

Comment:
Is this next test meant to capture a case where there is a second pod that would cause us to filter out reserved capacity options so we don't schedule it but then we get a third pod that schedules to a completely different NodePool?